src/contracts/Protocol/QuestionStateController.sol

```diff
@@ -24,35 +25,34 @@ contract QuestionStateController is IQuestionStateController, Ownable, OnlyApi {
      * @param questionId The id of the question
      */
     function initializeQuestion(uint256 questionId, string calldata uri) public onlyApi {
-        state[questionId] = STATE.VOTING;
-        QuestionData memory _question = QuestionData({url: uri, totalVotes: 0, questionId: questionId, questionState: STATE.VOTING});
-        questionByState[questionId] = _question;
+        QuestionStats memory question;
+
+        question.questionId = questionId;
+        question.uri = uri;
+        question.totalVotes = 1;
+        question.questionState = STATE.VOTING;
+
+        questionByState[questionId] = question;
     }
 
     function publish(uint256 questionId) public onlyApi onlyState(STATE.VOTING, questionId) {
         // if some voting barrier is passed, we can publish the question
-        state[questionId] = STATE.PUBLISHED;
+        QuestionStats storage _question = questionByState[questionId];
+        _question.questionState = STATE.PUBLISHED;
     }
 
-    function voteFor(
-        address _user,
-        uint256 questionId,
-        uint256 amount
-    ) public onlyApi onlyState(STATE.VOTING, questionId) {
+    function voteFor(address _user, uint256 questionId) public onlyApi onlyState(STATE.VOTING, questionId) {
         // Checks
         if (hasVoted[_user][questionId]) revert HasAlreadyVotedForQuestion();
 
         // Effects
-        Vote memory _vote = Vote({voter: _user, amount: amount});
-        votes[questionId].votes.push(_vote);
+        QuestionStats storage _question = questionByState[questionId];
+        _question.totalVotes += 1;
 
         hasVoted[_user][questionId] = true;
-        questionIndex[_user][questionId] = votes[questionId].votes.length - 1;
-
-        votes[questionId].totalVoteCount += amount; // TODO Lock tokens for voting include safeTransferFrom
+        _question.voters.push(_user);
+        questionIndex[_user][questionId] = _question.voters.length - 1;
```

Comment:
I guess I'm sort of confused on the purpose of this or what this does

Reply:
yep!